dot/sync/chain_sync_test.go

```diff
@@ -827,3 +827,140 @@ func TestChainSync_determineSyncPeers(t *testing.T) {
 	require.Equal(t, []peer.ID{testPeerB}, peers)
 }
+
+func TestChainSync_highestBlock(t *testing.T) {
+	type input struct {
+		peerState map[peer.ID]*peerState
+	}
+	type output struct {
+		highestBlock int64
+		err          error
+	}
+	type test struct {
+		name string
+		in   input
+		out  output
+	}
+	tests := []test{
+		{
+			name: "when has an empty map should return 0, errNoPeers",
+			in: input{
+				peerState: map[peer.ID]*peerState{},
+			},
+			out: output{
+				highestBlock: int64(0),
+				err:          errNoPeers,
+			},
+		},
+		{
+			name: "when has a nil map should return 0, errNoPeers",
+			in: input{
+				peerState: nil,
+			},
+			out: output{
+				highestBlock: int64(0),
+				err:          errNoPeers,
+			},
+		},
+		{
+			name: "when has only one peer with number 90 should return 90, nil",
+			in: input{
+				peerState: map[peer.ID]*peerState{
+					"idtest": {number: big.NewInt(90)},
+				},
+			},
+			out: output{
+				highestBlock: int64(90),
+				err:          nil,
+			},
+		},
+		{
+			name: "when has only one peer with number nil should return 0, errNilBlockData",
+			in: input{
+				peerState: map[peer.ID]*peerState{
+					"idtest": {number: nil},
+				},
+			},
+			out: output{
+				highestBlock: int64(0),
+				err:          errNilBlockData,
+			},
+		},
+		{
+			name: "when has two peers (p1, p2) with p1.number 90 and p2.number 190 should return 190, nil",
+			in: input{
+				peerState: map[peer.ID]*peerState{
+					"idtest#1": {number: big.NewInt(90)},
+					"idtest#2": {number: big.NewInt(190)},
+				},
+			},
+			out: output{
+				highestBlock: int64(190),
+				err:          nil,
+			},
+		},
+		{
+			name: "when has two peers (p1, p2) with p1.number 190 and p2.number 90 should return 190, nil",
+			in: input{
+				peerState: map[peer.ID]*peerState{
+					"idtest#1": {number: big.NewInt(190)},
+					"idtest#2": {number: big.NewInt(90)},
+				},
+			},
+			out: output{
+				highestBlock: int64(190),
+				err:          nil,
+			},
+		},
+		{
+			name: "when has two peers (p1, p2) with p1.number nil and p2.number 90 should return 90, nil",
+			in: input{
+				peerState: map[peer.ID]*peerState{
+					"idtest#1": {number: nil},
+					"idtest#2": {number: big.NewInt(90)},
+				},
+			},
+			out: output{
+				highestBlock: int64(90),
+				err:          nil,
+			},
+		},
+		{
+			name: "when has two peers (p1, p2) with p1.number 90 and p2.number nil should return 90, nil",
+			in: input{
+				peerState: map[peer.ID]*peerState{
+					"idtest#1": {number: big.NewInt(90)},
+					"idtest#2": {number: nil},
+				},
+			},
+			out: output{
+				highestBlock: int64(90),
+				err:          nil,
+			},
+		},
+		{
+			name: "when has two peers (p1, p2) with p1.number nil and p2.number nil should return 0, errNilBlockData",
+			in: input{
+				peerState: map[peer.ID]*peerState{
+					"idtest#1": {number: nil},
+					"idtest#2": {number: nil},
+				},
+			},
+			out: output{
+				highestBlock: int64(0),
```

Comment:
nit I think this should do (same for other test cases)

```suggestion
				highestBlock: 0,
```

Reply:
totally true. Done in 5bb3b1b